pkg/repository/v1_repository.go

```diff
@@ -542,3 +560,33 @@ func (r *V1Repository) FetchComponent(id string) (com *v1manifest.Component, err
 
 	return r.updateComponentManifest(id)
 }
+
+// BinaryPath return the binary path of the component.
+// Support you have install the component, need to get entry from local manifest.
+// Load the manifest locally only to get then Entry, do not force do something need access mirror.
+func (r *V1Repository) BinaryPath(installPath string, componentID string, version v0manifest.Version) (string, error) {
```

Comment:
Can we take a version string or something more generic than the v0 version?

Reply:
Actually I think this information should be stored locally rather than reproduced from scratch. It is possible the manifest might change but not installed location, or vice-versa. I think `Profile` should keep this information and it should be independent of whether we use version 0 or 1 manifests.

Reply:
agree, but like v0, it needs to read the manifest to get the `entry` to construct the path.
and we need to still work for the user upgrade from v0. Can only keep reading manifest before the profile support keeps this information.

Reply:
But this won't work with an upgrade from v0 because there won't be an index manifest, etc? I think this will also fail for v1 manifests if we download a new component manifest, but not the component itself